tests/zfs-tests/tests/functional/channel_program/lua_core/tst.encryption.ksh

```diff
@@ -0,0 +1,42 @@
+#!/bin/ksh -p
+# SPDX-License-Identifier: CDDL-1.0
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2021 by Determinate Systems. All rights reserved.
+#
+
+. $STF_SUITE/tests/functional/channel_program/channel_common.kshlib
+
+#
+# DESCRIPTION:
+#       zfs.exists should accurately report whether a dataset exists, and
+#       report an error if a dataset is in another pool.
+
+verify_runnable "global"
+
+TESTDATASET="channelprogramencryption"
+
+passphrase="password"
+log_must eval "echo "$passphrase" | zfs create -o encryption=aes-256-ccm " \
+        "-o keyformat=passphrase $TESTPOOL/$TESTDATASET"
+
+function cleanup
```

Comment:
Shouldn't somewhere here be `log_onexit cleanup`, preferably before doing anything?

Reply:
@amotin looking at other `tst.*.ksh` files in `lua_core` dir, I think this happened because almost all the other tests don't do this either, except `lua_core/tst.return_large.ksh`

shall I add another commit on top, which will add it to the rest of the files in `lua_core` too?

Reply:
That's the one I looked and rest was just a quick grep. Will fix these two right away.

Reply:
seems ok now:

```
cleanup.ksh:20:default_cleanup
tst.encryption.ksh:33:function cleanup
tst.encryption.ksh:38:log_onexit cleanup
tst.exists.ksh:31:function cleanup
tst.exists.ksh:36:log_onexit cleanup
tst.return_large.ksh:29:function cleanup
tst.return_large.ksh:34:log_onexit cleanup
```

Reply:
Thanks. But I think it may have sense to set up cleanup before creating the dataset to not leak it if the script is somehow aborted in between.